src/Silex/CallbackResolver.php

```diff
@@ -33,35 +33,45 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            if (!isset($this->app[$service])) {
+                throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            }
+
+            return array($this->app[$service], $method);
+        }
+
+        $callback = $this->app[$name];
+
+        if (!is_callable($callback)) {
+            throw new \InvalidArgumentException(sprintf('Service "%s" is not callable.', $name));
         }
 
-        return array($this->app[$service], $method);
+        return $callback;
     }
 
     /**
      * Returns a callable given its string representation if it is a valid service method.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return string|callable A callable value or the string passed in
```

Comment:
I think I'm confused by the previous `@return array A callable array` since any invalid value would be returned as is. But I guess the point is to only return a callable, right?
If I understand correctly we should either write `@return callable` or `@return mixed`.